### DISCUSSION TITLE
Custom enum codec

### DISCUSSION THREAD

**Discussion by soundgarddden** [MorphiaOrg/morphia#1726] [Q&A] [answered]
Hi
Did I understand correctly that it is not possible to add your codec for enum classes to Morphia mapper? Will dev.morphia.mapping.codec.EnumCodec always be used?

**Comment by evanchooly** [comment, marked as answer]
There isn't a Morphia way to do that yet, no.  You can register your codecs with the driver itself (using a `CodecProvider`) but depending on how it's mapped, Morphia might override that.  If the codec is for a specific enum type (rather than just "is it an enum?") then I think you should be fine.  2.3 is introducing an experimental way to introduce codecs but that method is primarily built around mapping entities rather than basic types so I'm not sure that would help you much either.
